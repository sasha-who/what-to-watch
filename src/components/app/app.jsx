import React from "react";
import PropTypes from "prop-types";
import Main from "../main/main.jsx";

const App = ({title, genre, date, films, onTitleClick}) => (
  <Main
    title={title}
    genre={genre}
    date={date}
    films={films}
    onTitleClick={onTitleClick}
  />
);

App.propTypes = {
  title: PropTypes.string.isRequired,
  genre: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  films: PropTypes.arrayOf(
      PropTypes.shape({
        title: PropTypes.string.isRequired,
        cover: PropTypes.string.isRequired
      })
  ).isRequired,
  onTitleClick: PropTypes.func.isRequired
};

export default App;
