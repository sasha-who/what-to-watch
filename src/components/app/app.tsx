import * as React from "react";
import {Switch, Route, Router, Link} from "react-router-dom";
import {connect} from "react-redux";
import Loader from "react-loader-spinner";
import history from "../../history";
import {LoaderData, HttpStatus, AppRoute} from "../../const";
import {ActionCreator} from "../../reducer/app-state/app-state";
import {Operation as DataOperation} from "../../reducer/data/data";
import {Operation as UserOperation} from "../../reducer/user/user";
import {
  getFilms,
  getPromoFilm,
  getFilmsLoadState,
  getPromoFilmLoadState,
  getRequestStatus,
  getFilmComments,
  getCommentsLoadState,
  getFavoriteFilms,
  getFavoriteFilmsLoadState
} from "../../reducer/data/selectors";
import {
  getCurrentGenre,
  getFilmsCountToShow,
  getActiveFilm,
  getFilteredFilms,
  getSimilarFilms
} from "../../reducer/app-state/selectors";
import {
  getAuthorizationStatus,
  getAuthorizationData,
  getCommentPostStatus,
  getLoginError
} from "../../reducer/user/selectors";
import PrivateRoute from "../private-route/private-route";
import Main from "../main/main";
import DetailedFilmCard from "../detailed-card/detailed-card";
import ServerError from "../server-error/server-error";
import AuthorizationScreen from "../authorization-screen/authorization-screen";
import ReviewScreen from "../review-screen/review-screen";
import MyList from "../my-list/my-list";
import Player from "../player/player";
import withValidityCheck from "../../hocs/with-validity-check/with-validity-check";
import withPlayer from "../../hocs/with-player/with-player";
import {AuthorizationData, Film, Comment} from "../../types";

interface Props {
  authorizationStatus: string;
  authorizationData: AuthorizationData;
  promoFilm: Film;
  films: Film[];
  filteredFilms: Film[];
  activeFilm: Film;
  similarFilms: Film[];
  favoriteFilms: Film[];
  activeFilmComments: Comment[];
  currentGenre: string;
  filmsCountToShow?: number;
  isFilmsLoaded: boolean;
  isPromoFilmLoaded: boolean;
  isCommentsLoaded: boolean;
  requestStatus: number;
  isFavoriteFilmsLoaded: boolean;
  loginError?: number;
  login: () => void;
  onActiveFilmChange: () => void;
  onGenreChange: () => void;
  onFilmsCountToShowReset: () => void;
  onFilmsCountToShowIncrement?: () => void;
  loadFilmComments: () => void;
  postReview: () => void;
  commentPostStatus: () => void;
  onFavoriteStatusChange: () => void;
  loadFavoriteFilms: () => void;
}

const AuthorizationScreenWrapped = withValidityCheck(AuthorizationScreen);
const PlayerWrapped = withPlayer(Player);

class App extends React.PureComponent<Props, null> {
  render() {
    const {
      authorizationStatus,
      authorizationData,
      isCommentsLoaded,
      films,
      promoFilm,
      filteredFilms,
      similarFilms,
      activeFilmComments,
      currentGenre,
      filmsCountToShow,
      isFilmsLoaded,
      isPromoFilmLoaded,
      requestStatus,
      favoriteFilms,
      isFavoriteFilmsLoaded,
      loginError,
      commentPostStatus,
      onActiveFilmChange,
      onGenreChange,
      onFilmsCountToShowReset,
      onFilmsCountToShowIncrement,
      login,
      loadFilmComments,
      postReview,
      onFavoriteStatusChange,
      loadFavoriteFilms
    } = this.props;

    if (!isFilmsLoaded || !isPromoFilmLoaded) {
      return (
        <div style={LoaderData.STYLE}>
          <Loader
            type={LoaderData.TYPE}
            color={LoaderData.COLOR}
            width={LoaderData.HEIGHT}
            height={LoaderData.WIDTH}
          />
        </div>
      );
    }

    if (!(requestStatus >= HttpStatus.SUCCESS && requestStatus < HttpStatus.REDIRECT)) {
      return <ServerError requestStatus={requestStatus} />;
    }

    return (
      <Router
        history={history}
      >
        <Switch>
          <Route exact path={AppRoute.ROOT}>
            <Main
              authorizationStatus={authorizationStatus}
              authorizationData={authorizationData}
              films={films}
              promoFilm={promoFilm}
              currentGenre={currentGenre}
              filteredFilms={filteredFilms}
              filmsCountToShow={filmsCountToShow}
              onGenreChange={onGenreChange}
              onFilmsCountToShowReset={onFilmsCountToShowReset}
              onFilmsCountToShowIncrement={onFilmsCountToShowIncrement}
              loadFilmComments={loadFilmComments}
              onFavoriteStatusChange={onFavoriteStatusChange}
            />
          </Route>
          <Route exact path={AppRoute.FILM}
            render={(props) => (
              <DetailedFilmCard
                {...props}
                authorizationStatus={authorizationStatus}
                authorizationData={authorizationData}
                isCommentsLoaded={isCommentsLoaded}
                films={films}
                similarFilms={similarFilms}
                activeFilmComments={activeFilmComments}
                onActiveFilmChange={onActiveFilmChange}
                loadFilmComments={loadFilmComments}
                onFavoriteStatusChange={onFavoriteStatusChange}
              />
            )}
          />
          <Route exact path={AppRoute.LOGIN}>
            <AuthorizationScreenWrapped
              authorizationStatus={authorizationStatus}
              onAuthorizationFormSubmit={login}
              loginError={loginError}
            />
          </Route>
          <Route
            exact
            path={AppRoute.PLAYER}
            render={(props) => (
              <PlayerWrapped
                {...props}
                films={films}
              />
            )}
          />
          <PrivateRoute
            exact
            path={AppRoute.REVIEW}
            render={(props) => (
              <ReviewScreen
                {...props}
                films={films}
                authorizationData={authorizationData}
                authorizationStatus={authorizationStatus}
                postReview={postReview}
                commentPostStatus={commentPostStatus}
              />
            )}
          />
          <PrivateRoute
            exact
            path={AppRoute.MY_LIST}
            render={(props) => (
              <MyList
                {...props}
                favoriteFilms={favoriteFilms}
                isFavoriteFilmsLoaded={isFavoriteFilmsLoaded}
                loadFilmComments={loadFilmComments}
                loadFavoriteFilms={loadFavoriteFilms}
                authorizationData={authorizationData}
                authorizationStatus={authorizationStatus}
              />
            )}
          />
          <Route
            render={() => (
              <React.Fragment>
                <h1>
                  Error: 404. Page not found.
                </h1>
                <Link to={AppRoute.ROOT}>Go to main page</Link>
              </React.Fragment>
            )}
          />
        </Switch>
      </Router>
    );
  }
}

const mapStateToProps = (state) => ({
  authorizationStatus: getAuthorizationStatus(state),
  authorizationData: getAuthorizationData(state),
  activeFilm: getActiveFilm(state),
  currentGenre: getCurrentGenre(state),
  films: getFilms(state),
  filteredFilms: getFilteredFilms(state),
  filmsCountToShow: getFilmsCountToShow(state),
  similarFilms: getSimilarFilms(state),
  promoFilm: getPromoFilm(state),
  activeFilmComments: getFilmComments(state),
  isFilmsLoaded: getFilmsLoadState(state),
  isPromoFilmLoaded: getPromoFilmLoadState(state),
  isCommentsLoaded: getCommentsLoadState(state),
  requestStatus: getRequestStatus(state),
  commentPostStatus: getCommentPostStatus(state),
  favoriteFilms: getFavoriteFilms(state),
  isFavoriteFilmsLoaded: getFavoriteFilmsLoadState(state),
  loginError: getLoginError(state)
});

const mapDispatchToProps = (dispatch) => ({
  login(authorizationData) {
    dispatch(UserOperation.login(authorizationData));
  },
  loadFilmComments(filmId) {
    dispatch(DataOperation.loadActiveFilmComments(filmId));
  },
  loadFavoriteFilms() {
    dispatch(DataOperation.loadFavoriteFilms());
  },
  postReview(review, filmId) {
    dispatch(UserOperation.postReview(review, filmId));
  },
  onFavoriteStatusChange(filmId, status, isPromoFilm) {
    dispatch(DataOperation.changeFavoriteStatus(filmId, status, isPromoFilm));
  },
  onActiveFilmChange(film) {
    dispatch(ActionCreator.setActiveFilm(film));
  },
  onGenreChange(genre) {
    dispatch(ActionCreator.changeCurrentGenre(genre));
  },
  onFilmsCountToShowReset() {
    dispatch(ActionCreator.resetFilmsCountToShow());
  },
  onFilmsCountToShowIncrement() {
    dispatch(ActionCreator.incrementFilmsCountToShow());
  }
});

export {App};
export default connect(mapStateToProps, mapDispatchToProps)(App);
