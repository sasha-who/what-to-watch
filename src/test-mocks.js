const RELEASE = `2020`;
const RATING = 9;
const RATINGS_COUNT = 228;
const DIRECTOR = `Otto Bathurst`;
const PREVIEW = `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`;
const RUN_TIME = 140;

const DESCRRIPTION = `In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's
friend and protege. Gustave prides himself on providing first-class service to the hotel's guests,
including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's
lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief
suspect in her murder.`;

const TITLES = [
  `Fantastic Beasts: The Crimes of Grindelwald`,
  `Bohemian Rhapsody`,
  `Macbeth`,
  `Aviator`,
  `We need to talk about Kevin`,
  `What We Do in the Shadows`,
  `Revenant`,
  `Johnny English`
];

const COVERS = [
  `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  `img/bohemian-rhapsody.jpg`,
  `img/macbeth.jpg`,
  `img/aviator.jpg`,
  `img/we-need-to-talk-about-kevin.jpg`,
  `img/what-we-do-in-the-shadows.jpg`,
  `img/revenant.jpg`,
  `img/johnny-english.jpg`
];

const ACTORS = [
  `Cillian Murphy`,
  `Sam Neill`,
  `Helen McCrory`,
  `Paul Anderson`,
  `Annabelle Wallis`,
  `Iddo Goldberg`
];

export const GENRES = [
  `all genres`,
  `comedy`,
  `drama`,
  `melodrama`,
  `horror`,
  `crime`,
  `documentary`,
  `romance`,
  `thriller`
];

export const TABS_DATA = [
  `overview`,
  `details`,
  `reviews`
];

const review = {
  id: `0`,
  text: `I didn't find it amusing, and while I can appreciate the creativity,
  it's an hour and 40 minutes I wish I could take back.`,
  rating: 8,
  userName: `Kate Muir`,
  date: new Date(`December 25, 2019 01:00:00`)
};

export const films = TITLES.map((title, index) => {
  return ({
    id: index.toString(),
    title,
    cover: COVERS[index],
    poster: COVERS[index],
    preview: PREVIEW,
    genre: GENRES[index + 1] || GENRES[1],
    release: RELEASE,
    rating: RATING,
    ratingsCount: RATINGS_COUNT,
    description: DESCRRIPTION,
    director: DIRECTOR,
    actors: ACTORS,
    runTime: RUN_TIME,
    reviews: [review]
  });
});
