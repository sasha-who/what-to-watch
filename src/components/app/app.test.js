import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {INITIAL_FILMS_COUNT, Screen} from "../../const.js";
import {films, GENRES} from "../../test-mocks.js";
import {App} from "./app.jsx";

const mockStore = configureStore([]);

const store = mockStore({
  activeScreen: Screen.MAIN,
  activeFilm: films[0],
  currentGenre: GENRES[0],
  films,
  filteredFilms: films,
  filmsCountToShow: INITIAL_FILMS_COUNT,
  isPlayerActive: false
});

it(`Main screen should render correctly`, () => {
  const tree = renderer
    .create(
        <Provider store={store}>
          <App
            activeScreen={Screen.MAIN}
            activeFilm={films[0]}
            promoFilm={films[0]}
            films={films}
            filteredFilms={films}
            filmsCountToShow={INITIAL_FILMS_COUNT}
            currentGenre={GENRES[0]}
            similarFilms={films}
            isPlayerActive={false}
            onScreenChange={() => {}}
            onActiveFilmChange={() => {}}
            onGenreChange={() => {}}
            onFilmsCountToShowReset={() => {}}
            onFilmsCountToShowIncrement={() => {}}
            onPlayerStateChange={() => {}}
          />
        </Provider>, {
          createNodeMock: () => {
            return {};
          }
        }
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Card screen should render correctly`, () => {
  const tree = renderer
    .create(
        <Provider store={store}>
          <App
            activeScreen={Screen.CARD}
            activeFilm={films[0]}
            promoFilm={films[0]}
            films={films}
            filteredFilms={films}
            filmsCountToShow={INITIAL_FILMS_COUNT}
            currentGenre={GENRES[0]}
            similarFilms={films}
            isPlayerActive={false}
            onScreenChange={() => {}}
            onActiveFilmChange={() => {}}
            onGenreChange={() => {}}
            onFilmsCountToShowReset={() => {}}
            onFilmsCountToShowIncrement={() => {}}
            onPlayerStateChange={() => {}}
          />
        </Provider>, {
          createNodeMock: () => {
            return {};
          }
        }
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
