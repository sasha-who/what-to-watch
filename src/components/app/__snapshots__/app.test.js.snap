// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card screen should render correctly 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="The Grand Budapest Hotel"
        src="img/bg-the-grand-budapest-hotel.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <a
          href="/mylist"
          onClick={[Function]}
        >
          <div
            className="user-block__avatar"
          >
            <img
              alt="User avatar"
              height={63}
              src="https://4.react.pages.academyimg/1.png"
              width={63}
            />
          </div>
        </a>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="The Grand Budapest Hotel poster"
            height={327}
            src="img/the-grand-budapest-hotel-poster.jpg"
            width={218}
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Fantastic Beasts: The Crimes of Grindelwald
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              comedy
            </span>
            <span
              className="movie-card__year"
            >
              2020
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <a
              className="btn btn--play movie-card__button"
              href="/player/0"
              onClick={[Function]}
            >
              <svg
                height={19}
                viewBox="0 0 19 19"
                width={19}
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </a>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height={20}
                viewBox="0 0 19 20"
                width={19}
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Comedy
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Drama
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Melodrama
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Horror
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Crime
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Documentary
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Romance
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Thriller
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__image"
            href="/films/0"
            onClick={[Function]}
          >
            <video
              height={175}
              muted={true}
              poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width={280}
            />
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/0"
              onClick={[Function]}
            >
              Fantastic Beasts: The Crimes of Grindelwald
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__image"
            href="/films/1"
            onClick={[Function]}
          >
            <video
              height={175}
              muted={true}
              poster="img/bohemian-rhapsody.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width={280}
            />
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/1"
              onClick={[Function]}
            >
              Bohemian Rhapsody
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__image"
            href="/films/2"
            onClick={[Function]}
          >
            <video
              height={175}
              muted={true}
              poster="img/macbeth.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width={280}
            />
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/2"
              onClick={[Function]}
            >
              Macbeth
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__image"
            href="/films/3"
            onClick={[Function]}
          >
            <video
              height={175}
              muted={true}
              poster="img/aviator.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width={280}
            />
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/3"
              onClick={[Function]}
            >
              Aviator
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__image"
            href="/films/4"
            onClick={[Function]}
          >
            <video
              height={175}
              muted={true}
              poster="img/we-need-to-talk-about-kevin.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width={280}
            />
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/4"
              onClick={[Function]}
            >
              We need to talk about Kevin
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__image"
            href="/films/5"
            onClick={[Function]}
          >
            <video
              height={175}
              muted={true}
              poster="img/what-we-do-in-the-shadows.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width={280}
            />
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/5"
              onClick={[Function]}
            >
              What We Do in the Shadows
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__image"
            href="/films/6"
            onClick={[Function]}
          >
            <video
              height={175}
              muted={true}
              poster="img/revenant.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width={280}
            />
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/6"
              onClick={[Function]}
            >
              Revenant
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__image"
            href="/films/7"
            onClick={[Function]}
          >
            <video
              height={175}
              muted={true}
              poster="img/johnny-english.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width={280}
            />
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/7"
              onClick={[Function]}
            >
              Johnny English
            </a>
          </h3>
        </article>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          © 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;

exports[`Main screen should render correctly 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="The Grand Budapest Hotel"
        src="img/bg-the-grand-budapest-hotel.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <a
          href="/mylist"
          onClick={[Function]}
        >
          <div
            className="user-block__avatar"
          >
            <img
              alt="User avatar"
              height={63}
              src="https://4.react.pages.academyimg/1.png"
              width={63}
            />
          </div>
        </a>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="The Grand Budapest Hotel poster"
            height={327}
            src="img/the-grand-budapest-hotel-poster.jpg"
            width={218}
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Fantastic Beasts: The Crimes of Grindelwald
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              comedy
            </span>
            <span
              className="movie-card__year"
            >
              2020
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <a
              className="btn btn--play movie-card__button"
              href="/player/0"
              onClick={[Function]}
            >
              <svg
                height={19}
                viewBox="0 0 19 19"
                width={19}
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </a>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height={20}
                viewBox="0 0 19 20"
                width={19}
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Comedy
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Drama
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Melodrama
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Horror
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Crime
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Documentary
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Romance
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Thriller
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__image"
            href="/films/0"
            onClick={[Function]}
          >
            <video
              height={175}
              muted={true}
              poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width={280}
            />
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/0"
              onClick={[Function]}
            >
              Fantastic Beasts: The Crimes of Grindelwald
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__image"
            href="/films/1"
            onClick={[Function]}
          >
            <video
              height={175}
              muted={true}
              poster="img/bohemian-rhapsody.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width={280}
            />
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/1"
              onClick={[Function]}
            >
              Bohemian Rhapsody
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__image"
            href="/films/2"
            onClick={[Function]}
          >
            <video
              height={175}
              muted={true}
              poster="img/macbeth.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width={280}
            />
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/2"
              onClick={[Function]}
            >
              Macbeth
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__image"
            href="/films/3"
            onClick={[Function]}
          >
            <video
              height={175}
              muted={true}
              poster="img/aviator.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width={280}
            />
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/3"
              onClick={[Function]}
            >
              Aviator
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__image"
            href="/films/4"
            onClick={[Function]}
          >
            <video
              height={175}
              muted={true}
              poster="img/we-need-to-talk-about-kevin.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width={280}
            />
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/4"
              onClick={[Function]}
            >
              We need to talk about Kevin
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__image"
            href="/films/5"
            onClick={[Function]}
          >
            <video
              height={175}
              muted={true}
              poster="img/what-we-do-in-the-shadows.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width={280}
            />
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/5"
              onClick={[Function]}
            >
              What We Do in the Shadows
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__image"
            href="/films/6"
            onClick={[Function]}
          >
            <video
              height={175}
              muted={true}
              poster="img/revenant.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width={280}
            />
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/6"
              onClick={[Function]}
            >
              Revenant
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__image"
            href="/films/7"
            onClick={[Function]}
          >
            <video
              height={175}
              muted={true}
              poster="img/johnny-english.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width={280}
            />
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/7"
              onClick={[Function]}
            >
              Johnny English
            </a>
          </h3>
        </article>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          © 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
